### ISSUE TITLE
About the training with prompt_image_alignment configuration which uses llava_bertscore reward function

### ISSUE THREAD

**Issue by QZJ-2003** [kvablack/ddpo-pytorch#11]
The result of training with llava_bertscore reward is:
![image](https://github.com/kvablack/ddpo-pytorch/assets/75468610/24931895-be01-41c1-b0bb-058d7f4ad525)
But the result of training with aesthetic reward is:
![image](https://github.com/kvablack/ddpo-pytorch/assets/75468610/471d167c-d4bc-4378-b164-2c6fadd49c6f)
This shows that the reinforcement learning works. 
The configuration file, i.e. `dgx.py` and `base.py`, were not modified.
The hardware used in the training is 8 A800. The llava-server ran in GPU0 and GPU1. The training program ran in all 8 GPUs.
GPU0 and GPU1 was used 59G memory, and others use 30G memory.
The version of llava is liuhaotian/LLaVA-Lightning-MPT-7B-preview.

**Comment by kvablack** [comment]
Sorry, I'm not really sure I can help you debug this, although I did use the 13B variant of LLaVA.

**Comment by kvablack** [comment]
I'll also mention that you should pull the recent changes to fix the bug with gradient synchronization, as this effectively multiplies the batch size by 8 if you're using 8 GPUs.

**Comment by Owen-Oertell** [comment]
I see this as well. Is there anything that we can do to try and solve this issue? Thanks. I am using the 13B variant of LLaVA. Further, this was working before I pulled the most recent changes.

**Comment by kvablack** [comment]
@Owen-Oertell That's a bit concerning. Could you post a reward curve from before and after commit 5955244f37d74295fa9959c3bef999b1ef1cf7b8, with all hyperparameters the same? Thanks.

**Comment by Owen-Oertell** [comment]
@kvablack I apologize for the delay. I actually was able to resolve the issue by increasing the number of GPUs.

**Comment by Dswn888** [comment]
> [@kvablack](https://github.com/kvablack) I apologize for the delay. I actually was able to resolve the issue by increasing the number of GPUs.

Hi @Owen-Oertell ,
Thank you for sharing your solution—it’s been very helpful. I’m currently facing a similar issue while training with 6 GPUs. Could you please provide more details on how increasing the GPU count resolved the problem, and whether there were any additional adjustments you made? Your further insights would be greatly appreciated.

Best regards,
David Sun
